### ISSUE TITLE
[BUG] After a while, the packet We send from the mqtt connector does not reach the thingsboard server.

### ISSUE THREAD

**Issue by gohm-tech** [thingsboard/thingsboard-gateway#658]
The thingsboard-gateway service stops sending data to the server after running for a certain period of time. When I restart the thingsboard-gateway service, it starts sending again.

That is the server code. The part marked in yellow is our gateway's acces token. We could not see any logs containing the acces token of our gateway when it did not send the data to the server. After restarting the thingsboard-gateway service, these logs started to appear.
![Screenshot from 2021-12-20 08-50-54](https://user-images.githubusercontent.com/91528436/146919542-949f066c-12f6-45ba-86b0-56e17e672918.png)

That is our code log.
![Screenshot from 2021-12-21 13-48-09](https://user-images.githubusercontent.com/91528436/146918439-c610ad38-968a-4ae6-8a76-3d4efe1f612e.png)

That is connector log.
![Screenshot from 2021-12-21 13-47-57](https://user-images.githubusercontent.com/91528436/146918479-807c3d92-c96d-4c0e-b7d4-706f30667ffc.png)

That is the MQTT broker log we connect and subscribe.
![Screenshot from 2021-12-21 13-53-34](https://user-images.githubusercontent.com/91528436/146918680-728e58d3-6c08-4096-95ba-1a55573523da.png)

**Server Environment**
own setup
docker deployment
OS: Linux(Ubuntu 20.04)

**Client Environment**
OS: Linux(Ubuntu)
Browser: Chrome, Firefox
Version Ubuntu Version 20.04

**Connector name**
MQTT Connector

**Configuration**
[mqtt.txt](https://github.com/thingsboard/thingsboard-gateway/files/7752196/mqtt.txt)
[tb_gateway.txt](https://github.com/thingsboard/thingsboard-gateway/files/7752203/tb_gateway.txt)

**Versions:**
OS: Raspbian GNU/Linux 10 (buster)
Thingsboard IoT Gateway version [e.g. 2.7]
Python version 3.7.3

Best Regards,
**Gohm Team**

**Comment by samson0v** [comment]
Hi @gohm-tech, please, set DEBUG level in the `config/logs.conf` file and provide more info.
Thanks for your interest in ThingsBoard IoT Gateway!

**Comment by gohm-tech** [comment]
Hi @samson0v,
We already set DEBUG level the all of the log levels. We did not see any error in the log files. Everythings look good.Best Regards,
**Gohm Team**

**Comment by samson0v** [comment]
@gohm-tech, after about how long do Gateway stop sending data?

**Comment by gohm-tech** [comment]
Hi @samson0v,
We test it again. There are the logs.
[service.log](https://github.com/thingsboard/thingsboard-gateway/files/7762201/service.log)
[connector.log](https://github.com/thingsboard/thingsboard-gateway/files/7762202/connector.log)
[thingsboard_server.log](https://github.com/thingsboard/thingsboard-gateway/files/7762207/thingsboard_server.log)
![1](https://user-images.githubusercontent.com/91528436/147083240-3192eac7-08aa-4a1e-9a43-46abdbe41a22.png)
![2](https://user-images.githubusercontent.com/91528436/147083249-d6f9889e-8e5e-4277-88fb-7bfa8138efd0.png)
![3](https://user-images.githubusercontent.com/91528436/147083309-5bb0385e-1144-4804-8ada-c253f0126e26.png)

**Note:** The server clock is 3 hours behind.

**Comment by gohm-tech** [comment]
Hi @samson0v, 

Do you think it can be related to our thingsboard-gateway version?

Regards,
**Gohm Team**

**Comment by gohm-tech** [comment]
Hi @samson0v,

We created two new gateways. One of them has the 2.7 version and the other has the latest version. (The reason we use version 2.7 is high CPU usage in the latest version). We have created a service that will send data over an MQTT connector per second. We think that this way we can understand whether the error is caused by the version.

Have you encountered such an error before?
Do you have any ideas as to why this error might occur?

Regards,
**Gohm Team**

**Comment by samson0v** [comment]
Hi @gohm-tech, we released a new version of Gateway 3.0 (include a High CPU usage fix), so I think it will be better if you **completely** delete all versions of Gateways you have, install the new (3.0) one and send the logs from it.
Thanks for your interest in ThingsBoard IoT Gateway!

**Comment by gohm-tech** [comment]
Hi @samson0v,

Thanks for your interest. Thanks for this information too. After the testing we realized something. thingsboard-gateway and this code are working together. We have created a service for working continuously to mqtt_client_firmware.py code. This code wants to subscribe same topics. So we were getting this error #643. After getting this error we change the ACTORS_MAX_CONCURRENT_SESSION_PER_DEVICE parameter. After doing this we thought the error was resolved. But in this test environment, we disabled the service of mqq_client_firmware.py code. After doing this, the error did not occur. In short, the error occurs as a result of multiple MQTT subscribe.

Regards,
**Gohm Team**
